cmd/wrestic/integration_test.go

```diff
@@ -0,0 +1,251 @@
+// +build integration
+
+package main
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"context"
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"git.vshn.net/vshn/wrestic/s3"
+
+	"git.vshn.net/vshn/wrestic/output"
+	"git.vshn.net/vshn/wrestic/restic"
+)
+
+type webhookserver struct {
+	jsonData []byte
+	srv      *http.Server
+}
+
+type testEnvironment struct {
+	s3Client *s3.Client
+	output   *output.Output
+	webhook  *webhookserver
+	finishC  chan error
+	t        *testing.T
+}
+
+func assertOK(t *testing.T, err error) {
```

Comment:
A nice library for asserts in Go, FYI: https://godoc.org/github.com/stretchr/testify/assert

Reply:
Thanks for the info.

For the current state that library is a bit overkill. But I'll keep it in mind for implementing future tests.